File: 1992. Find All Groups of Farmland/Solution.cpp
#include <functional>
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> findFarmland(vector<vector<int>>& land)
    {
        int m = land.size(), n = land[0].size();
        vector<vector<int>> res;

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (land[i][j] != 1)
                    continue;

                int r = i, c = j;
                while (r + 1 < m && land[r + 1][j] == 1)
                    r++;
                while (c + 1 < n && land[i][c + 1] == 1)
                    c++;
                res.push_back({ i, j, r, c });

                for (int x = i; x <= r; x++)
                    for (int y = j; y <= c; y++)
                        land[x][y] = 2;
            }
        }
        return res;
    }
};

File: 1992. Find All Groups of Farmland/Solution.py
from typing import List


class Solution:
    def findFarmland(self, land: List[List[int]]) -> List[List[int]]:
        m, n = len(land), len(land[0])
        res = []

        for i in range(m):
            for j in range(n):
                if land[i][j] != 1:
                    continue

                r, c = i, j
                while r + 1 < m and land[r + 1][j] == 1:
                    r += 1
                while c + 1 < n and land[i][c + 1] == 1:
                    c += 1
                res.append([i, j, r, c])

                for x in range(i, r + 1):
                    for y in range(j, c + 1):
                        land[x][y] = 2
        return res
